packages/flutter/lib/src/rendering/binding.dart

```diff
@@ -143,10 +143,12 @@ mixin RendererBinding on BindingBase, ServicesBinding, SchedulerBinding, Gesture
   ///
   /// Called automatically when the binding is created.
   void initRenderView() {
-    assert(renderView == null);
+    assert(!_isRenderViewInitialized);
+    _isRenderViewInitialized = true;
     renderView = RenderView(configuration: createViewConfiguration(), window: window);
-    renderView!.prepareInitialFrame();
+    renderView.prepareInitialFrame();
   }
+  bool _isRenderViewInitialized = false;
```

Comment:
if we only use this for the assert, then call it "_debugIsRenderViewInitialized" and set it in the assert itself. that way it can be tree-shaken out of release builds.

Reply:
(and more importantly, it's debug-only status is then self-documenting.)

Reply:
Done